js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
+
+// TokenState values
+const (
+	ExprState TokenState = iota
+	StmtParensState
+	SubscriptState
+	PropNameState
+)
```

Comment:
*shrug* I don't care much if you can think of a better name, just chose what was used to.